aws/rust-runtime/aws-http/src/retry.rs

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+//! AWS-specific retry logic
+
+use aws_smithy_http::result::SdkError;
+use aws_smithy_http::retry::ClassifyResponse;
+use aws_smithy_types::retry::{ErrorKind, ProvideErrorKind, RetryKind};
+use std::time::Duration;
+
+/// A retry policy that models AWS error codes as outlined in the SEP
+///
+/// In order of priority:
+/// 1. The `x-amz-retry-after` header is checked
+/// 2. The modeled error retry mode is checked
+/// 3. The code is checked against a predetermined list of throttling errors & transient error codes
+/// 4. The status code is checked against a predetermined list of status codes
+#[non_exhaustive]
+#[derive(Clone, Debug)]
+pub struct AwsErrorRetryPolicy;
+
+const TRANSIENT_ERROR_STATUS_CODES: &[u16] = &[500, 502, 503, 504];
+const THROTTLING_ERRORS: &[&str] = &[
+    "Throttling",
+    "ThrottlingException",
+    "ThrottledException",
+    "RequestThrottledException",
+    "TooManyRequestsException",
+    "ProvisionedThroughputExceededException",
+    "TransactionInProgressException",
+    "RequestLimitExceeded",
+    "BandwidthLimitExceeded",
+    "LimitExceededException",
+    "RequestThrottled",
+    "SlowDown",
+    "PriorRequestNotComplete",
+    "EC2ThrottledException",
+];
```

Comment:
Where do these come from? It could be educational to link to a reference doc here if possible